### PR TITLE
(#1216) Flex Layout Utility Page

### LINKED ISSUE #1216: Layout and Positioning subpage: Flex
Update the Flex page within the Layout and Positioning section of the documentation site with the content needed for NCIDS for All pilot launch.

### Sub-Tasks
- [ ] Draft content for page
- [ ] Add images (if needed)
- [ ] Convert to .mdx
- [ ] Update documentation site page

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #1216

## Pull Request Details

_Add description_

Closes #<TICKET-NUMBER>

## Author PR Checklist
Items that the author of the PR is responsible for checking before submitted the PR.

### General:
- [x] I have reviewed the acceptance criteria defined in the ticket and ensured the work has been completed.
- [x] The commit message passes [all quality commit message standards](https://github.com/NCIOCPL/ncids/wiki/Creating-Quality-Commit-Messages).
- [x] Unit tests have been updated or created to reflect any javascript changes.
- [x] Storybook scenarios have been updated or created to reflect any html/css/js changes.

### Accessibility:
- [x] WCAG 2.1 Level AA requirements have been met.

### Development:
- [x] Any new or updated javascript code has 100% unit test coverage.
- [x] New or updated breakpoints have regression images.
- [x] Breaking changes have been thoroughly documented in the PR.

## Product Reviewer PR Checklist
Items the product team is responsible for reviewing.

### General:
- [x] There are no unexpected or unapproved regression image changes. 
- [x] Functionality of interactive elements meet the acceptance criteria.
- [x] The product is visually and functionally the same across the different browsers.

### Accessibility:
- [x] AxeDev Tools: there are no new or outstanding accessibility issues introduced in this PR.
- [x] Lighthouse: scores have not noticeably decreased during this PR.
- [x] Wave: there are no new errors or contrast errors introduced in this PR.

## Design Reviewer PR Checklist
Items the design team is responsible for reviewing.  

### General:
- [x] New or updated features introduced in this PR are developed mobile-first.
- [x] Breakpoint changes and regression images match those breakpoints.
- [x] This PR has been tested in all supported browsers at all breakpoints.

## Developer Reviewer PR Checklist
Items the development team is responsible for reviewing.

### General:
- [ ] New code passes code quality standards set by industry standards.
- [ ] The expected Storybook stories have been added or updated for the new or updated feature.
- [ ] The expected unit tests have been added or updated for the new or updated feature.

### Accessibility:
- [ ] VoiceOver: Described content matches with what was expected.
- [ ] Keyboard navigation: new or updated features and content are navigable via the keyboard.

### CODE REVIEW

**Review by sarinapadilla — APPROVED** [review]
Looks good to me! I'm seeing all of the utility modules from the [spreadsheet](https://nih.sharepoint.com/:x:/r/sites/NCI-OCPLNewCgovUXDesignSystem/_layouts/15/doc2.aspx?sourcedoc=%7BAEAC153A-696A-4419-941B-142CC9B627F1%7D&file=USWDS_Utilities.xlsx&action=default&mobileredirect=true) and all of the examples are displaying correctly.

File: docs/content/foundations/layout-positioning/flex.mdx [hunk 1]
```diff
@@ -13,6 +13,282 @@ utility_modules:
...
+    utility_examples:
+      - heading:
+        description: |
+        preview: true
+        code: |
+          <section className="utility-examples">
+            <div className="grid-row grid-gap-sm margin-bottom-1">
+              <div className="grid-col flex-1">
+                <div className="height-6 width-full display-flex flex-align-center bg-secondary-light flex-justify-center padding-x-105">
+                  <span className="site-utility-examples__class text-no-wrap">.flex-1</span>
```
Comment (olitharp-nci) [inline docs/content/foundations/layout-positioning/flex.mdx:25]: please remove uswds-site classes.

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by olitharp-nci — CHANGES_REQUESTED** [review]
You should not be showing presentation code like this on the live site. What is shown to the user should only be code relevant for them to copy and paste.

This is bad:

Screenshot 2024-07-30 at 10 57 31 AM

This is what is expected: 

Screenshot 2024-07-30 at 10 59 35 AM

Rework both this pr and #1803.

Also please create a ticket to go back and fix float as this detail was missed. Please have product review again.

File: docs/content/foundations/layout-positioning/flex.mdx [hunk 2]
```diff
@@ -10,56 +10,383 @@ utility_modules:
...
+        - class_name: 'flex-11'
+        - class_name: 'flex-12'
+        - class_name: 'flex-auto'
+        - class_name: 'flex-full'
+    utility_examples:
+      - heading:
+        description: |
+        preview: false
+        code: |
+          <section>
```
Comment (olitharp-nci) [inline docs/content/foundations/layout-positioning/flex.mdx:37]: remove the section wrappers

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Comment by olitharp-nci** [comment]
Stop resolving conversations. Let the PR reviewer do it.

**Comment by olitharp-nci** [comment]
Has content been reviewed? It doesn't seem that "The flex utility classes have not been modified from the U.S. Web Design System (USWDS). Refer to the USWDS Flex utility page for a complete listing of classes" is needed and it's repeated for each item.

File: docs/content/foundations/layout-positioning/flex.mdx (hunk 2 above)
Comment (olitharp-nci) [reply docs/content/foundations/layout-positioning/flex.mdx]: okie this looks good now

**Review by olitharp-nci — APPROVED** [review]

**Review by andyvanavery31 — APPROVED** [review]
Looks good to me!

**Pull request merged**